hack/ci-deploy-release.sh

```diff
@@ -0,0 +1,109 @@
+    exit 1
+fi
+
+ARTIFACTS="${ARTIFACTS:-$( mktemp -d )}"
+
+kubectl_create -n=prometheus-operator -f="${source_url}/${revision}/examples/third-party/prometheus-operator.yaml"
+kubectl_create -n=haproxy-ingress -f="${source_url}/${revision}/examples/third-party/haproxy-ingress.yaml"
+
+kubectl_create -f="${source_url}/${revision}/examples/common/cert-manager.yaml"
+# Wait for cert-manager crd and webhooks
+kubectl wait --for condition=established --timeout=60s crd/certificates.cert-manager.io crd/issuers.cert-manager.io
+for d in cert-manager{,-cainjector,-webhook}; do
+    kubectl -n=cert-manager rollout status --timeout=5m "deployment.apps/${d}"
+done
+wait-for-object-creation cert-manager secret/cert-manager-webhook-ca
+
+mkdir "${ARTIFACTS}/operator"
+cat > "${ARTIFACTS}/operator/kustomization.yaml" << EOF
+resources:
+- ${source_url}/${revision}/deploy/operator.yaml
+patches:
+- patch: |-
+    apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      name: scylla-operator
+      namespace: scylla-operator
+    spec:
+      template:
+        spec:
+          containers:
+          - name: scylla-operator
+            image: "${operator_image_ref}"
+            env:
+            - name: SCYLLA_OPERATOR_IMAGE
+              value: "${operator_image_ref}"
+EOF
+kubectl kustomize "${ARTIFACTS}/operator" | kubectl_create -n=scylla-operator -f=-
+
+# Manager needs scylla CRD registered and the webhook running
+kubectl wait --for condition=established crd/{scyllaclusters,nodeconfigs}.scylla.scylladb.com
+kubectl -n=scylla-operator rollout status --timeout=5m deployment.apps/scylla-operator
+kubectl -n=scylla-operator rollout status --timeout=5m deployment.apps/webhook-server
+
+if [[ -z "${SO_NODECONFIG_PATH:-}" ]]; then
+ echo "Skipping NodeConfig creation"
+else
+ kubectl_create -f="${SO_NODECONFIG_PATH}"
+fi
+
+kubectl_create -n=local-csi-driver -f="${source_url}/${revision}/examples/common/local-volume-provisioner/local-csi-driver/"{00_namespace.yaml,00_scylladb-local-xfs.storageclass.yaml,10_csidriver.yaml,10_driver.serviceaccount.yaml,10_provisioner_clusterrole.yaml,20_provisioner_clusterrolebinding.yaml,50_daemonset.yaml}
+kubectl -n=local-csi-driver rollout status --timeout=5m daemonset.apps/local-csi-driver
+
+mkdir "${ARTIFACTS}/manager"
+cat > "${ARTIFACTS}/manager/kustomization.yaml" << EOF
+resources:
+- ${source_url}/${revision}/deploy/manager-prod.yaml
+patches:
+- target:
+    group: scylla.scylladb.com
+    version: v1
+    kind: ScyllaCluster
+    name: scylla-manager-cluster
+  patch: |
+    - op: replace
+      path: /spec/datacenter/racks/0/storage/storageClassName
+      value: scylladb-local-xfs
+EOF
+kubectl kustomize "${ARTIFACTS}/manager" | kubectl_create -n=scylla-manager -f=-
+
+wait-for-object-creation scylla-manager statefulset.apps/scylla-manager-cluster-manager-dc-manager-rack
+kubectl -n=scylla-manager rollout status --timeout=5m statefulset.apps/scylla-manager-cluster-manager-dc-manager-rack
```

Comment:
nit: these could be replaced with waiting for our three aggregated conditions on ScyllaCluster without hardcoding names of Operator managed resources.

Reply:
It's a good point but it's in ci-deploy.sh as well, I'll send a followup.
/hold cancel

Reply:
https://github.com/scylladb/scylla-operator/pull/1947